### PR TITLE
Align `error!` method signatures across different places.

### PR DESCRIPTION
**Pull request by numbata**
This pull request fix a compatibility issue with the `error!` method when used in `rescue_from` block.

Previously, the `rescue_from` block ran in context of `Grape::Middleware::Error`, where `error!` method had one signature. Now, block run in context of `Grape::DSL::InsideRoute`, where `error!` method has different signature. This change cause runtime errors when `error!` method was called with extra arguments like `backtrace` and `original_exception`.

This update make `error!` method signatures same across different contexts. By aligning these signatures, `error!` method can handle additional arguments consistently and preventing errors.

### CODE REVIEW

**Comment by numbata** [comment]
The [2eabc9b](https://github.com/ruby-grape/grape/pull/2468/commits/2eabc9bfe2415cf3891b24e81a5c503a8ad99b30) is on top the commit before the 2377 PR was merged. And it passes now.

**Review by dblock — APPROVED** [review]
Looks good!

**Comment by numbata** [comment]
@dblock any plans for release this?

**Comment by dblock** [comment]
I can do it this week unless @ericproulx or another maintainer wants to beat me to it?

**Comment by numbata** [comment]
Just a quick reminder!
The Renovate bot is eager to update our dependencies (including grape), but we're holding off until these changes will be published.

**Comment by dblock** [comment]
Andrei, it will get done when it will get done. Your company is welcome to hire any of the maintainers of grape to do it on your schedule, or at least [sponsor the project](https://tidelift.com/funding/github/rubygems/grape).

**Comment by numbata** [comment]
> Andrei, it will get done when it will get done.

No pressure! And I would be happy to hire or sponsor any or all of the maintainers, but I hope I can contribute in other ways as well. By my PR's for example.

**Comment by numbata** [comment]
Off topic about sponsoring the project: I was looking for an easy way to enter my credit card information and set up a monthly "buy a couple of coffees for grape maintainers" donation. But it's not very clear to me where to do that on the Tidelift page. So I just gave up.

**Comment by dblock** [comment]
Appreciate it. I do have a [sponsor profile](https://github.com/sponsors/dblock) :)

**Comment by ericproulx** [comment]
@numbata Grape 2.1.3 is out :) Thank you for your patience.

**Pull request merged**
